openhands/runtime/utils/runtime_build.py

```diff
@@ -22,55 +21,49 @@ def get_runtime_image_repo():
 def _put_source_code_to_dir(temp_dir: str):
     """Builds the project source tarball directly in temp_dir and unpacks it.
     The OpenHands source code ends up in the temp_dir/code directory.
-
     Parameters:
     - temp_dir (str): The directory to put the source code in
     """
     if not os.path.isdir(temp_dir):
         raise RuntimeError(f'Temp directory {temp_dir} does not exist')
 
-    project_root = os.path.dirname(os.path.dirname(os.path.abspath(openhands.__file__)))
-    logger.info(f'Building source distribution using project root: {project_root}')
-
-    # Fetch the correct version from pyproject.toml
-    package_version = oh_version
-    tarball_filename = f'openhands_ai-{package_version}.tar.gz'
-    tarball_path = os.path.join(temp_dir, tarball_filename)
-
-    # Run "python -m build -s" on project_root to create project tarball directly in temp_dir
-    _cleaned_project_root = project_root.replace(
-        ' ', r'\ '
-    )  # escape spaces in the project root
-    result = subprocess.run(
-        f'python -m build -s -o "{temp_dir}" {_cleaned_project_root}',
-        shell=True,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-    )
-    logger.info(result.stdout.decode())
-    err_logs = result.stderr.decode()
-    if err_logs:
-        logger.error(err_logs)
-
-    if result.returncode != 0:
-        logger.error(f'Image build failed:\n{result}')
-        raise RuntimeError(f'Image build failed:\n{result}')
-
-    if not os.path.exists(tarball_path):
-        logger.error(f'Source distribution not found at {tarball_path}')
-        raise RuntimeError(f'Source distribution not found at {tarball_path}')
-    logger.info(f'Source distribution created at {tarball_path}')
-
-    # Unzip the tarball
-    shutil.unpack_archive(tarball_path, temp_dir)
-    # Remove the tarball
-    os.remove(tarball_path)
-    # Rename the directory containing the code to 'code'
-    os.rename(
-        os.path.join(temp_dir, f'openhands_ai-{package_version}'),
-        os.path.join(temp_dir, 'code'),
-    )
-    logger.info(f'Unpacked source code directory: {os.path.join(temp_dir, "code")}')
+    package_name = 'openhands-ai'
+    dest_dir = os.path.join(temp_dir, 'code')
+
+    try:
+        # Try to get the source directory from the installed package
+        distribution = importlib.metadata.distribution(package_name)
+        source_dir = os.path.dirname(distribution.locate_file(package_name))
+        openhands_dir = os.path.join(source_dir, 'openhands')
```

Comment:
If this fails before, wouldn't `openhands_dir` be an undefined variable?

Reply:
Will add a sanity check.

Reply:
Sounds good! I think once this sanity test is added, this PR should be ready to merge

Reply:
I revisited this code and you are right that if the package doesn't exist (which happens when running from source code or packaged docker image), `openhands_dir` would be undefined.

The next code section would check if `openhands_dir` exists or not. If not, then it uses the current working directory as source (almost same as current behavior except it doesn't use the "build" system).